components/login-screen.tsx

```diff
@@ -1,60 +1,113 @@
 import React, {useState} from 'react';
-import {View, Text, TextInput, Button, StyleSheet} from 'react-native';
+import {View, Text, TextInput, Button, StyleSheet, Alert} from 'react-native';
+import {
+  ApolloClient,
+  InMemoryCache,
+  HttpLink,
+  useMutation,
+  gql,
+  ApolloProvider,
+} from '@apollo/client';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
+const client = new ApolloClient({
+  link: new HttpLink({
+    uri: 'https://template-onboarding-node-sjz6wnaoia-uc.a.run.app/graphql',
+  }),
+  cache: new InMemoryCache(),
+});
+
+const LOGIN_MUTATION = gql`
+  mutation Mutation($data: LoginInput!) {
+    login(data: $data) {
+      token
+    }
+  }
+`;
 
 const LoginScreen = () => {
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
   const [emailError, setEmailError] = useState('');
   const [passwordError, setPasswordError] = useState('');
+  const [login, {loading}] = useMutation(LOGIN_MUTATION, {
+    client: client,
+  });
 
-  const handleLogin = () => {
-    const emailTrimmed = email.trim();
-    const passwordTrimmed = password.trim();
+  const handleLogin = async () => {
+    setEmailError(email.trim() ? '' : 'Insira seu e-mail.');
+    setPasswordError(password.trim() ? '' : 'Insira sua senha.');
 
-    setEmailError(emailTrimmed ? '' : 'Insira seu e-mail.');
-    setPasswordError(passwordTrimmed ? '' : 'Insira sua senha.');
-
-    if (emailTrimmed && passwordTrimmed) {
-      if (!/^\S+@\S+\.\S+$/.test(emailTrimmed)) {
+    if (email.trim() && password.trim()) {
+      if (!/^\S+@\S+\.\S+$/.test(email.trim())) {
         setEmailError('Formato do e-mail inválido.');
         return;
       }
 
       if (
         password.length < 7 ||
-        !/\d/.test(passwordTrimmed) ||
-        !/[a-zA-Z]/.test(passwordTrimmed)
+        !/\d/.test(password.trim()) ||
+        !/[a-zA-Z]/.test(password.trim())
```

Comment:
Nit: é improvável, porém possível que alguém tenha um espaço ao início ou ao final da senha, então eu evitria dar trim aqui.